src/main/java/net/minecraftforge/event/world/WorldEvent.java

```diff
@@ -118,4 +124,44 @@ public ServerLevelData getSettings()
             return settings;
         }
     }
+
+    /**
+     * Called by NaturalSpawner to gather a list of all possible entities that can spawn at the specified location.
+     * If an entry is added to the list, it needs to be a globally unique instance.
+     * The event is called in {@link NaturalSpawner#getRandomSpawnMobAt()}.
+     * Canceling the event will result in an empty list, meaning no entity will be spawned.
+     */
+    @net.minecraftforge.eventbus.api.Cancelable
+    public static class PotentialSpawns extends WorldEvent
+    {
+        private final MobCategory type;
+        private final BlockPos pos;
+        private final List<MobSpawnSettings.SpawnerData> list;
+
+        public PotentialSpawns(LevelAccessor world, MobCategory type, BlockPos pos, WeightedRandomList<MobSpawnSettings.SpawnerData> oldList)
+        {
+            super(world);
+            this.pos = pos;
+            this.type = type;
+            if (!oldList.isEmpty())
+              this.list = new ArrayList<>(oldList.unwrap());
```

Comment:
Formatting: 4 spaces for indentation.
```suggestion
                this.list = new ArrayList<>(oldList.unwrap());
```